modules/dop/services/issue/validate.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issue
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/dop/dao"
+)
+
+type issueValidator struct {
+	iteration *dao.Iteration
+	state     *dao.IssueState
+	db        *dao.DBClient
```

Comment:
it quite like `cache`, can you separate `cache` logic out of validator?
abstract a cache-flow.

Reply:
Changed to use in memory cache